src/System.CommandLine.Hosting/HostingExtensions.cs

```diff
@@ -108,6 +108,12 @@ public static IHostBuilder UseCommandHandler(this IHostBuilder builder, Type com
                 && command.GetType() == commandType)
             {
                 invocation.BindingContext.AddService(handlerType, c => c.GetService<IHost>().Services.GetService(handlerType));
+                var baseType = handlerType.BaseType;
+                while (baseType != null && baseType != typeof(object))
+                {
+                    invocation.BindingContext.AddService(baseType, c => c.GetService<IHost>().Services.GetService(handlerType));
```

Comment:
If I use two different derived types this way, will one not overwrite the other's base type registration?

Reply:
Very true. Was a bit too short-sighted when the code got rid of the initial error sorry for that :/
I have explored an other option, though:
1. adding a overload for `CommandHandler.Create(...)` that takes the concrete handler type (called inside `UseCommandHandler`)
2. adding similar overloads all the way to `ModelBindingCommandHandler`
3. using the concrete type inside of `ModelBindingCommandHandler` to resolve the handler from the concrete type , falling back to the previously used `_handlerMethodInfo!.DeclaringType` if no concrete type was specified (here `_handlerMethodInfo!.DeclaringType` is the base-type, that implements `ICommandHandler.InvokeAsync`)

I think we should ultimately get rid of this fallback behavior and require the handler type to be specified. Even if a overload with a concrete `_invocationTarget` was used, since it could be null.
Didn't do that yet, since it comes with breaking changes on some lower level classes (`.UseCommandHandler()` signature would not change) and a bunch of calls needing changes.
So I wanted to hear what you have to say about this before going forward.

Should i add the new code here? Or make a  new PR?